### PR TITLE
Added SynonymValidator

### PR DESCRIPTION
**Pull request by assertchris**
Cinnamon validator. Hmmm...

### CODE REVIEW

**Comment by tractorcow** [comment]
This looks good, but maybe we should put this in the CWP module instead of the open source one? Without the cwp core api this doesn't actually have any use case.

**Comment by assertchris** [comment]
I dunno. It does seem reasonable to see this if non-CWP applications want CMS-controlled synonyms...

File: tests/SynonymValidateTest.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			array('foo, bar ,baz'),
+			array('
+				foo
+				bar
+				baz
+			'),
+			array('
+				# this is a comment, it should be ignored!
+
+				foo=>bar,baz
```
Comment (tractorcow) [inline tests/SynonymValidateTest.php]: I think we need to support `foo => bar, baz`. Also we can only have one `=>` per line :D

**Comment by tractorcow** [comment]
There's no CMS control for synonyms in this module though.

**Comment by assertchris** [comment]
I see that. The feature is tied to enforcing valid Solr configuration values, and can be used without being linked to a form (as the unit tests demonstrate). I don't mind where it goes, but I see merit in it living in either place.

File: tests/SynonymValidateTest.php (hunk 1 above)
Comment (assertchris) [inline tests/SynonymValidateTest.php]: ok!

**Comment by tractorcow** [comment]
Can you put it with the SynonymsSiteConfig class in cwp? You'll need a PR to that module anyway in order to ensure the validator is enabled. :D

**Comment by assertchris** [comment]
Ok, moved to https://gitlab.cwp.govt.nz/cwp/cwp/merge_requests/82.

**Pull request closed without merging**